### DISCUSSION TITLE
Understanding dataflow in MMDetection

### DISCUSSION THREAD

**Discussion by saheban** [open-mmlab/mmengine#1325] [Q&A]
Hello and thanks for providing this helpful library,
I have a question about dataflow which I can't find where I should manipulate the dataflow in order to have my own dataflow?
Let me elaborate what I mean.
As you know in a detection network and in training process, first, we pass the input tensor to the backbone which may be a ResNet or other network, after that, it is turn to pass the outputs of backbone to a neck like FPN, and outputs of neck to detector for classification and regression and finally calculation loss and update whole the model.
So here is my question, this process of passing input tensor and outputs of each step to another step is happening in which file and in which file object of related classes is defined?

**Comment by zhouzaida** [comment]
Hi, we currently do not have the specific data flow in MMDetection, but we have a generic flow about MMEngine (https://user-images.githubusercontent.com/58739961/253929306-267db9cb-72e4-4af2-a58b-877b30091acc.png).
